Kitodo/src/main/java/org/kitodo/services/data/TaskService.java

```diff
@@ -700,6 +705,35 @@ private void abortTask(Task task) throws DataException {
         save(task);
     }
 
+    /**
+     * Performs creating images when this happens automatically in a task.
+     *
+     * @param executingThread
+     *            Executing thread (displayed in the taskmanager)
+     * @param task
+     *            Task that generates images
+     * @param automatic
+     *            Whether it is an automatic task
+     * @throws DataException
+     *             If the task cannot be saved
+     */
+    public void generateImages(EmptyTask executingThread, Task task, boolean automatic) throws DataException {
+        Process process = task.getProcess();
+        ImageGenerator generator = new ImageGenerator(new Subfolder(process, process.getProject().getGeneratorSource()),
+                GenerationMode.ALL, SubfolderFactoryService.createAll(process, task.getContentFolders()));
+        generator.setSupervisor(executingThread);
+        if (automatic) {
+            if (executingThread.getException() == null) {
+                task.setEditType(TaskEditType.AUTOMATIC.getValue());
```

Comment:
In both cases the same TaskEditType is set up. In that case it should be set before if.

Reply:
You're right, that could have been better cleaned up, and I did that now. The function is now implemented only once and runs in script tasks and generate images equally.